src/Servers/Kestrel/test/Interop.FunctionalTests/Http2/Http2RequestTests.cs

```diff
@@ -37,6 +42,94 @@ public async Task GET_NoTLS_Http11RequestToHttp2Endpoint_400Result()
         }
     }
 
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public async Task GET_RequestReturnsLargeData_GracefulShutdownDuringRequest_RequestGracefullyCompletes(bool hasTrailers)
+    {
+        // Arrange
+        const int DataLength = 500_000;
+        var randomBytes = Enumerable.Range(1, DataLength).Select(i => (byte)((i % 10) + 48)).ToArray();
+
+        var syncPoint = new SyncPoint();
+
+        ILogger logger = null;
+        IHostApplicationLifetime hostApplicationLifetime = null;
+        var builder = CreateHostBuilder(
+            async c =>
+            {
+                await syncPoint.WaitToContinue();
+
+                var memory = c.Response.BodyWriter.GetMemory(randomBytes.Length);
+
+                logger.LogInformation($"Server writing {randomBytes.Length} bytes response");
+                randomBytes.CopyTo(memory);
+
+                logger.LogInformation($"Server advancing {randomBytes.Length} bytes response");
+                c.Response.BodyWriter.Advance(randomBytes.Length);
```

Comment:
Can we add a comment indicating that not awaiting the response body writes is important for the regression test?